client/server/middleware/unsupported-browser.js

```diff
@@ -18,13 +18,21 @@ function isSupportedBrowser( req ) {
 	} );
 }
 
+// We don't want to redirect some of our public landing pages, so we include them
+// here.
+function shouldRedirectPath( path ) {
+	const allowedPaths = [ 'browsehappy', 'log-in', 'start', 'new', 'themes' ];
+	return ! allowedPaths.includes( path.slice( 1 ) );
```

Comment:
Could `path` could contain a leading `/`?

Reply:
The test should catch URLs like `/themes/travel` or `/theme/twenty-twenty-one`. Therefore:
```js
[ '/browsehappy', '/log-in', ..., '/themes', '/theme' ].some( p => path.startsWith( p ) )
```

And there are also localized theme pages served by Calypso with URLs like `/de/themes` or `/es/theme/foo`.

Reply:
> Could path contain a leading /?

Yep, the leading '/' is removed.

I've also updated the matcher with @jsnajdr's suggestions:
- uses `startsWith`
- strips out leading locale using `config( 'magnificent_non_en_locales' )`

As an example, an input of `/es/theme/twenty-twenty` will be parsed to `theme/twenty-twenty`, and since that starts with `theme`, it will be allowed.